src/lib/comms/sol-oic-cbor.c

```diff
@@ -63,7 +63,7 @@ initialize_cbor_payload(struct sol_oic_map_writer *encoder)
 
     cbor_encoder_init(&encoder->encoder, encoder->payload,
         buf->capacity - offset, 0);
-    old_ptr = encoder->encoder.ptr;
+    old_ptr = encoder->encoder.data.ptr;
```

Comment:
This is the first time I update a git submodule but it seems ok in the patch, a088996aa5f59b4f27f20fadad053d88bee357d4 is the commit id of tinycbor v0.4: 
```
diff --git a/src/thirdparty/tinycbor b/src/thirdparty/tinycbor
index 629d5b7..a088996 160000
--- a/src/thirdparty/tinycbor
+++ b/src/thirdparty/tinycbor
@@ -1 +1 @@
-Subproject commit 629d5b71823ad91faab60f3f51597f010ad98c0c
+Subproject commit a088996aa5f59b4f27f20fadad053d88bee357d4
```
It seems git will not explicitly reference/show the v0.4 tag.